### PR TITLE
Yauheni/76938/timeline ts migration

### PR DESCRIPTION
**Pull request by yauheni-deriv**
## Changes:

Please include a summary of the change and which issue is fixed below:
-   ...

## When you need to add unit test

-   [ ] If this change disrupt current flow
-   [ ] If this change is adding new flow

## When you need to add integration test

-   [ ] If components from external libraries are being used to define the flow, e.g. @deriv/components
-   [ ] If it relies on a very specific set of props with no default behavior for the current component.

## Test coverage checklist (for reviewer)

-   [ ] Ensure utility / function has a test case 
-   [ ] Ensure all the tests are passing

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [x] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: packages/components/src/components/timeline/timeline.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    return (
+        <div className='dc-timeline__oval'>
+            <Text color='colored-background' size='s' weight='bold' className='dc-timeline__number'>
+                {children}
+            </Text>
+        </div>
+    );
+};
+
+type TItem = React.HTMLAttributes<HTMLDivElement> & { item_title: string | React.ReactNode };
```
Comment (niloofar-deriv) [inline packages/components/src/components/timeline/timeline.tsx]: `string` is a part of `ReactNode` types, do we need to define the `string` again?

File: packages/components/src/components/timeline/timeline.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                        className={classNames('dc-timeline__flex', {
+                            'dc-timeline__flex--no-border': children.length === idx + 1,
+                            'dc-timeline__flex--disabled': disabled_items && disabled_items.includes(idx + 1),
+                        })}
+                    >
+                        <Oval>{idx + 1}</Oval>
+                        <div className='dc-timeline__container'>
+                            <Text as='h2' color='prominent' size='xs' className='dc-timeline__title'>
+                                {child.props.item_title}
+                            </Text>
```
Comment (amina-deriv) [inline packages/components/src/components/timeline/timeline.tsx:37]: can we change this to {child?.props?.item_title} ?

File: packages/components/src/components/timeline/timeline.tsx (hunk 1 above)
Comment (yauheni-kryzhyk-deriv) [reply packages/components/src/components/timeline/timeline.tsx:19]: yeah, but there is no item_title value in this type, so I had to add it

File: packages/components/src/components/timeline/timeline.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            <Text color='colored-background' size='s' weight='bold' className='dc-timeline__number'>
+                {children}
+            </Text>
+        </div>
+    );
+};
+
+type TItem = React.HTMLAttributes<HTMLDivElement> & { item_title: string | React.ReactNode };
+
+type TTimeline = React.HTMLAttributes<HTMLDivElement> & { disabled_items: number[] };
```
Comment (maryia-deriv) [inline packages/components/src/components/timeline/timeline.tsx]: @yauheni-kryzhyk-deriv I suggest moving all type declarations to the top of the file🙂

File: packages/components/src/components/timeline/timeline.tsx (hunk 1 above)
Comment (yauheni-deriv) [reply packages/components/src/components/timeline/timeline.tsx]: @niloo-fs thanks, will check it

File: packages/components/src/components/timeline/timeline.tsx (hunk 2 above)
Comment (yauheni-deriv) [reply packages/components/src/components/timeline/timeline.tsx:37]: as I see in the code, we pass title everywhere, So I believe this prop should be necessary.
if it is necessary, we should stay without question marks, how do you think?

**Review by maryia-deriv — APPROVED** [review]

File: packages/components/src/components/timeline/timeline.tsx (hunk 2 above)
Comment (amina-deriv) [reply packages/components/src/components/timeline/timeline.tsx:37]: Thank you for the clarification. @yauheni-kryzhyk-deriv

**Review by amina-deriv — APPROVED** [review]

**Review by niloofar-deriv — APPROVED** [review]
LGTM

**Review by shayan-deriv — APPROVED** [review]
LGTM

**Pull request merged**